### PR TITLE
Stream policy updates to GRPC clients

### PR DESCRIPTION
**Pull request by wazir-ahmed**
- Added new GPRC service to stream policy updates
- The service will stream both KubeArmor and Cilium policy updates.

### CODE REVIEW

File: src/networkpolicy/consumer.go [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+		policy = plugin.ConvertKnoxNetworkPolicyToCiliumPolicy(*knoxNetPolicy)
+	}
+	if policyType == dpb.PolicyType_K8S_NETWORK {
+		// TODO: Handle this after k8s network policy support is added
+		return nil
+	}
+
+	jsonBytes, err := json.Marshal(&policy)
+	if err != nil {
+		log.Error().Msgf("CiliumNetworkPolicy json marshal failed err=%v", err.Error())
```
Comment (nyrahul) [inline src/networkpolicy/consumer.go]:
```suggestion
		log.Error().Msgf("Network Policy json marshal failed err=%v", err.Error())
```

**Review by seswarrajan — APPROVED** [review]
LGTM.

**Review by seswarrajan — APPROVED** [review]
LGTM. @nyrahul Please provide your comments.

**Comment by seswarrajan** [comment]
@wazir-ahmed 
Please remove lines as well, since the cluster_name and container_name is removed from source.
https://github.com/kubearmor/discovery-engine/blob/797e6b4fbd027ef0938a9d5a4ec6b9c26c9cd9be/src/systempolicy/systemPolicy.go#L326

**Comment by wazir-ahmed** [comment]
> 



> @wazir-ahmed Please remove lines as well, since the cluster_name and container_name is removed from source.
> 
> https://github.com/kubearmor/discovery-engine/blob/797e6b4fbd027ef0938a9d5a4ec6b9c26c9cd9be/src/systempolicy/systemPolicy.go#L326

@seswarrajan Fixed this as well.

File: src/libs/consumer.go [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+	resp := convertPolicyYamlToGrpcResponse(policy)
+	err := stream.SendMsg(resp)
+	if err != nil {
+		log.Error().Msgf("sending network policy yaml in grpc stream failed err=%v", err.Error())
+		return err
+	}
+	return nil
+}
+
+func RelayPolicyEventToGrpStream(stream grpc.ServerStream, consumer *PolicyConsumer) error {
```
Comment (nyrahul) [inline src/libs/consumer.go]:
```suggestion
func RelayPolicyEventToGrpcStream(stream grpc.ServerStream, consumer *PolicyConsumer) error {
```

File: src/protobuf/v1/discovery/discovery.proto [hunk 3]
```diff
@@ -0,0 +1,26 @@
+syntax = "proto3";
+
+package v1.discovery;
+
+option go_package = "github.com/accuknox/knoxAutoPolicy/protos/v1/discovery";
```
Comment (nyrahul) [inline src/protobuf/v1/discovery/discovery.proto]: fix the package path. github.com/kubearmor/discovery-engine

**Review by nyrahul — APPROVED** [review]
LGTM. Few minor comments.

File: src/networkpolicy/consumer.go (hunk 1 above)
Comment (wazir-ahmed) [reply src/networkpolicy/consumer.go]: I think, you were checking an old commit. This line doesn't exist in the PR.

File: src/libs/consumer.go (hunk 2 above)
Comment (wazir-ahmed) [reply src/libs/consumer.go]: Fixed.

File: src/protobuf/v1/discovery/discovery.proto (hunk 3 above)
Comment (wazir-ahmed) [reply src/protobuf/v1/discovery/discovery.proto]: Fixed.

**Pull request merged**